pkg/collect/host_httploadbalancer.go

```diff
@@ -0,0 +1,150 @@
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"path"
+	"runtime/debug"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/segmentio/ksuid"
+)
+
+func HostHTTPLoadBalancer(c *HostCollector) (map[string][]byte, error) {
+	defer func() {
+		if r := recover(); r != nil {
+			log.Println(string(debug.Stack()))
+		}
+	}()
+	listenAddress := fmt.Sprintf("0.0.0.0:%d", c.Collect.HTTPLoadBalancer.Port)
+
+	timeout := 60 * time.Minute
+	if c.Collect.HTTPLoadBalancer.Timeout != "" {
+		var err error
+		timeout, err = time.ParseDuration(c.Collect.HTTPLoadBalancer.Timeout)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to parse timeout %q", c.Collect.HTTPLoadBalancer.Timeout)
+		}
+	}
+
+	requestToken := ksuid.New().Bytes()
+	responseToken := ksuid.New().Bytes()
+
+	go func() {
+		mux := http.NewServeMux()
+		server := http.Server{
+			Addr:    listenAddress,
+			Handler: mux,
+		}
+
+		mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+			if r.Method != "POST" {
+				return
+			}
+			body, err := ioutil.ReadAll(r.Body)
+			if err != nil {
+				return
+			}
+			if !bytes.Equal(body, requestToken) {
+				return
+			}
+			_, err = w.Write(responseToken)
+			if err != nil {
+				return
+			}
+			server.Shutdown(context.Background())
+		})
+
+		http.ListenAndServe(listenAddress, mux)
+	}()
+
+	var networkStatus NetworkStatus
+
+	stopAfter := time.Now().Add(timeout)
+	for {
+		if time.Now().After(stopAfter) {
+			break
+		}
+
+		networkStatus = attemptPOST(c.Collect.HTTPLoadBalancer.Address, requestToken, responseToken)
+		if networkStatus == NetworkStatusErrorOther || networkStatus == NetworkStatusConnectionTimeout {
+			time.Sleep(50 * time.Millisecond)
+			continue
+		}
+
+		break
+	}
+
+	result := NetworkStatusResult{
+		Status: networkStatus,
+	}
+
+	b, err := json.Marshal(result)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal result")
+	}
+
+	name := path.Join("httpLoadBalancer", "httpLoadBalancer.json")
+	if c.Collect.HTTPLoadBalancer.CollectorName != "" {
+		name = path.Join("httpLoadBalancer", fmt.Sprintf("%s.json", c.Collect.HTTPLoadBalancer.CollectorName))
+	}
+
+	return map[string][]byte{
+		name: b,
+	}, nil
+}
+
+func attemptPOST(address string, request []byte, response []byte) NetworkStatus {
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+	defer cancel()
+
+	// Create a new transport every time to ensure a new TCP connection so the load balancer does
+	// not forward every request to the same backend
+	transport := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   100 * time.Millisecond,
+			DualStack: true,
+		}).DialContext,
+		ForceAttemptHTTP2: true,
+	}
+	client := http.Client{
+		Transport: transport,
+	}
+
+	buf := bytes.NewBuffer(request)
+	req, err := http.NewRequestWithContext(ctx, "POST", address, buf)
+	if err != nil {
+		fmt.Println(err.Error())
```

Comment:
This would be a bug such as context is nil, which is why this one is logged but no other errors are.